### PR TITLE
Misc release notes

### PR DESCRIPTION
**Pull request by Carreau**
Misc addition to release notes.
I've also tagged all the PRs merged since 2.4 with the 2.5 milestone.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

File: docs/release.rst [hunk 1]
```diff
@@ -14,6 +17,8 @@ Next release
 * Fix minor bug in `N5Store`. 
   By :user:`gsakkis`, :issue:`550`.
 
+* add new `FSStore` to open any ``fsspec`` supported protocol :issue:`546`
+
```
Comment (jakirkham) [inline docs/release.rst:21]: Looks like this actually shows up below. So maybe we can drop it?

**Review by jakirkham — APPROVED** [review]
After looking at the updated notes, it appears we have everything here. Multiget appears listed below and so does the removal of error functions. Maybe I was looking at an outdated diff before? Anyways this looks good on my end. Thanks Matthias! 😄

File: docs/release.rst (hunk 1 above)
Comment (Carreau) [reply docs/release.rst:21]:
```suggestion
```

**Comment by Carreau** [comment]
Travis CI appear twice (one green, one orange) and the orange one seem to be a bug as te URL is the same as the passing one. 
I'll override and merge.

**Comment by jakirkham** [comment]
Interesting was wondering if this was the leftover `travis-ci.org` project, but going to that link it shows the project as migrated to `travis-ci.com`. Not sure why the other build showed up.

Visiting the associated pending Travis CI detail link shows it as passing as well. So it may just be the GitHub status API didn't work when updating the build as passed.

Anyways all this to say ignoring it seems fine and I also don't have any clue why it showed up 😅

**Comment by Carreau** [comment]
it's not the first time I see that. So 🤷

**Pull request merged**